src/v/archival/archival_metadata_stm.cc

```diff
@@ -773,20 +773,26 @@ ss::future<bool> archival_metadata_stm::do_sync(
     // The log eviction STM can also replicate its own batch without holding the
     // lock but it's not critical.
 
-    auto insync = _manifest->get_insync_offset();
+    //
+    auto barrier_result = co_await _raft->linearizable_barrier();
```

Comment:
This is wrong. A replicate command might still live in the batcher and linearizable barrier can get an offset before that in-flight replicate...